core/src/lib/bsys.h

```diff
@@ -65,4 +65,12 @@ bool PathAppend(PoolMem& path, PoolMem& extra);
 bool PathCreate(const char* path, mode_t mode = 0750);
 bool PathCreate(PoolMem& path, mode_t mode = 0750);
 
+struct StateFileHeader {
+  char id[14];
+  int32_t version;
+  uint64_t last_jobs_addr;
+  uint64_t end_of_recent_job_results_list;
+  uint64_t reserved[19];
+};
+
```

Comment:
As this struct is used file-mapped (and we already ran into issues with it) we should check that the members are in locations we expect. With `static_assert` this is a simple compile-time check.

```suggestion

static_assert(offsetof(StateFileHeader, version) == 16);
static_assert(offsetof(StateFileHeader, last_jobs_addr) == 20 ||
                      offsetof(StateFileHeader, last_jobs_addr) == 24);
static_assert(offsetof(StateFileHeader, end_of_recent_job_results_list) == 28 ||
                      offsetof(StateFileHeader, end_of_recent_job_results_list) == 32);
static_assert(offsetof(StateFileHeader, reserved) == 36 ||
                      offsetof(StateFileHeader, reserved) == 40);
```